### PR TITLE
SpuVmSeqKeyOff etc

### PR DESCRIPTION
**Pull request by sozud**

### CODE REVIEW

File: src/main/psxsdk/libsnd/vmanager.c [hunk 1]
```diff
@@ -11,7 +11,47 @@ INCLUDE_ASM("main/nonmatchings/psxsdk/libsnd/vmanager", SpuVmDoAllocate);
 
 INCLUDE_ASM("main/nonmatchings/psxsdk/libsnd/vmanager", vmNoiseOn);
 
-INCLUDE_ASM("main/nonmatchings/psxsdk/libsnd/vmanager", vmNoiseOn2);
+void vmNoiseOn2(u8 arg0, u16 arg1, u16 arg2, u16 arg3, u16 arg4) {
```
Comment (Xeeynamo) [inline src/main/psxsdk/libsnd/vmanager.c:14]: I simplified the function a bit: https://decomp.me/scratch/8n4mp

Also I think `arg0` can become `voice`? and the existing `voice` can become `vc`, `i`, `index` or something similar?

File: src/main/psxsdk/libsnd/vmanager.c [hunk 2]
```diff
@@ -291,7 +331,39 @@ s16 SpuVmGetSeqRVol(s16 arg0) {
...
 }
 
-INCLUDE_ASM("main/nonmatchings/psxsdk/libsnd/vmanager", SpuVmSeqKeyOff);
+void SpuVmSeqKeyOff(s16 arg0) {
+    u8 voice;
+    s32 bitsUpper;
+    s32 bitsLower;
+    u16 temp2;
+    voice = 0;
+    if (spuVmMaxVoice != 0) {
```
Comment (Xeeynamo) [inline src/main/psxsdk/libsnd/vmanager.c:340]: This can be converted as a `for`

File: src/main/psxsdk/libsnd/vmanager.c [hunk 3]
```diff
@@ -528,7 +600,42 @@ s16 SsUtKeyOnV(s16 voice, s16 vabId, s16 prog, s16 tone, s16 note, s16 fine,
     return voice;
 }
 
-INCLUDE_ASM("main/nonmatchings/psxsdk/libsnd/vmanager", SsUtKeyOffV);
+s32 SsUtKeyOffV(u16 arg0) {
+    s32 bitsUpper;
+    s32 bitsLower;
+    u16 temp2;
+
+    if (_snd_ev_flag != 1) {
```
Comment (Xeeynamo) [inline src/main/psxsdk/libsnd/vmanager.c:608]: This looks like using two guard clauses, which will save nesting

File: src/main/psxsdk/libsnd/vmanager.c [hunk 4]
```diff
@@ -642,4 +749,53 @@ s16 SsUtAutoPan(s16 vc, s16 start_pan, s16 end_pan, s16 delta_time) {
...
 
-INCLUDE_ASM("main/nonmatchings/psxsdk/libsnd/vmanager", SsUtAllKeyOff);
+void SsUtAllKeyOff(void) {
+    s16 voice;
+    s32 bitsUpper;
+    s32 bitsLower;
+    u16 temp2;
+    s16 pos;
+    voice = 0;
+    if (spuVmMaxVoice != 0) {
```
Comment (Xeeynamo) [inline src/main/psxsdk/libsnd/vmanager.c:759]: `for` loop here

File: src/main/psxsdk/libspu/s_m_f.c [hunk 5]
```diff
@@ -1,3 +1,38 @@
...
+            temp_v1 = cur_mem->addr;
+
+            if (!(temp_v1 & mask4)) {
+                cur_block_num += 1;
+                if (temp_v1 == arg0) {
+                    cur_mem->addr = temp;
+                } else {
+                    cur_mem++;
+                    if (cur_block_num < block_num) {
+                        continue;
```
Comment (Xeeynamo) [inline src/main/psxsdk/libspu/s_m_f.c:30]: If you want to remove that `continue` and avoid some nesting, you can rewrite the `while` loop like here: https://decomp.me/scratch/XwfQs

The last `break;` is a bit weird. I tried to convert it into a `for` or a `do{} while` but unsuccessfully.

**Review by Xeeynamo — APPROVED** [review]
I left some suggestions, but this is good to be in

File: src/main/psxsdk/libsnd/vmanager.c (hunk 2 above)
Comment (sozud) [reply src/main/psxsdk/libsnd/vmanager.c:340]: Not getting a match here https://decomp.me/scratch/VSzmz

Comment (Xeeynamo) [reply src/main/psxsdk/libsnd/vmanager.c:340]: You need to remove the `if`: https://decomp.me/scratch/U2Zir

**Pull request merged**